var/spack/repos/builtin/packages/neurodamus/package.py

```diff
@@ -83,41 +94,78 @@ def syn2_dep_libs(self):
-                                              spec['hdf5'].prefix.include,
-                                              profile_flag)
-            link_flag += ' %s -L%s -lhdf5 -L%s -lz' % (
-                         spec['reportinglib'].libs.ld_flags,
-                         spec['hdf5'].prefix.lib,
-                         spec['zlib'].prefix.lib)
-
-            nrnivmodl = which('nrnivmodl')
+    def build(self, spec, prefix):
+        """ Build mod files from m dir
+        """
+        profile_flag = '-DENABLE_TAU_PROFILER' if '+profile' in spec else ''
+        include_flag = ''
+        link_flag = ''
+        env['MAKEFLAGS'] = '-j{0}'.format(make_jobs)
+
+        if '+syntool' in spec:
+            include_flag += ' -DENABLE_SYNTOOL -I ' + spec['synapsetool'].prefix.include
+            link_flag += self.syntool_dep_libs()
+
+        if '+coreneuron' in spec:
+            include_flag += ' -DENABLE_CORENEURON -I%s' % (spec['coreneuron'].prefix.include)
+            link_flag += ' %s' % (spec['coreneuron'].libs.ld_flags)
+
+        include_flag += ' -I%s -I%s %s' % (spec['reportinglib'].prefix.include,
+                                           spec['hdf5'].prefix.include,
+                                           profile_flag)
+        link_flag += ' %s -L%s -lhdf5 -L%s -lz' % (
+                     spec['reportinglib'].libs.ld_flags,
+                     spec['hdf5'].prefix.lib,
+                     spec['zlib'].prefix.lib)
+
+        nrnivmodl = which('nrnivmodl')
+        with profiling_wrapper_on():
             nrnivmodl('-incflags', include_flag,
-                      '-loadflags', link_flag,
-                      modlib)
+                      '-loadflags', link_flag, 'm')
+        # special exists or fail
+        special = os.path.join(os.path.basename(self.neuron_archdir), 'special')
+        assert os.path.isfile(special)
+
+    def install(self, spec, prefix):
+        """ Move libs to destination.
+            Libs are sym-linked. Compiled libs into libs, special into bin
+        """
+        neurodamus_base = spec['neurodamus-base'].prefix
+        arch = os.path.basename(self.neuron_archdir)
+        os.makedirs(prefix.lib.modc)
+        os.makedirs(prefix.bin)
+
+        os.symlink(neurodamus_base.lib.hoclib, prefix.lib.hoclib)
+        os.symlink(neurodamus_base.lib.modlib, prefix.lib.modlib)
+
+        if os.path.isdir(neurodamus_base.python):
+            os.symlink(neurodamus_base.python, prefix.python)
+
+        shutil.move(os.path.join(arch, 'special'), prefix.bin)
+
+        # Copy c mods
+        for cmod in find(arch, "*.c", recursive=False):
+            shutil.move(cmod, prefix.lib.modc)
 
-    @run_after('install')
-    def check_install(self):
-        special = '%s/special' % join_path(self.prefix, os.path.basename(self.neuron_archdir))
-        if not os.path.isfile(special):
-            raise RuntimeError("Neurodamus installion check failed, %s not created!" % special)
```

Comment:
I wonder if we should keep above check. This has been useful in multiple occasions for me (e.g. when nrnivmodl finish without error code)